nova/tests/unit/compute/test_api.py

```diff
@@ -2230,6 +2235,58 @@ def test_resize_same_host_and_allowed(self):
     def test_resize_same_host_and_not_allowed(self):
         self._test_resize(same_host=True, allow_same_host=False)
 
+    @ddt.data({'status': 'in-use', 'attach_status': 'attached',
+               'migration_status': None},
+              {'status': 'available', 'attach_status': 'detached',
+               'migration_status': None},
+              {'status': 'in-use', 'attach_status': 'attached',
+               'migration_status': 'success'},
+              {'status': 'available', 'attach_status': 'detached',
+               'migration_status': 'error'},)
+    def test_resize_with_volume_in_valid_state(self, fake_volume):
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)],)
+        fake_volume['id'] = uuids.volume_id
+        with mock.patch.object(self.compute_api.volume_api, 'get',
+                               return_value=fake_volume):
+            self._test_resize(bdms=bdms)
+
+    @ddt.data({'status': 'in-use', 'attach_status': 'detached',
+               'migration_status': None},
+              {'status': 'available', 'attach_status': 'attached',
+               'migration_status': None},
+              {'status': 'error', 'attach_status': 'detached',
+               'migration_status': None},
+              {'status': 'reserved', 'attach_status': 'detached',
+               'migration_status': None},
+              {'status': 'in-use', 'attach_status': 'attached',
+               'migration_status': 'migrating'},
+              {'status': 'available', 'attach_status': 'detached',
+               'migration_status': 'migrating'},)
+    def test_resize_with_volume_in_invalid_state(self, fake_volume):
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)])
+        fake_volume['id'] = uuids.volume_id
+        with mock.patch.object(self.compute_api.volume_api, 'get',
+                               return_value=fake_volume):
+            self.assertRaises(exception.InvalidVolume,
+                              self._test_resize, bdms=bdms)
+
+    def test_resize_with_2_volumes(self):
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id),
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)])
```

Comment:
These are 2 BDMs for the same Volume, which would be an inconsistent state in itself. Better `uuids.volume_id2` on the second..